pkg/operatorsource/configuring.go

```diff
@@ -15,21 +17,23 @@ import (
 
 // NewConfiguringReconciler returns a Reconciler that reconciles
 // an OperatorSource object in "Configuring" phase.
-func NewConfiguringReconciler(logger *log.Entry, datastore datastore.Writer, client client.Client) Reconciler {
-	return NewReconcilerWithInterfaceClient(logger, datastore, interface_client.NewClient(client))
+func NewConfiguringReconciler(logger *log.Entry, factory appregistry.ClientFactory, datastore datastore.Writer, client client.Client, refresher PackageRefreshNotificationSender) Reconciler {
+	return NewConfiguringReconcilerWithInterfaceClient(logger, factory, datastore, interfaceClient.NewClient(client), refresher)
 }
 
-// NewReconcilerWithInterfaceClient returns a configuring Reconciler
-// that reconciles an OperatorSource object in "Configuring" phase.
-// It uses the Client interface which is a wrapper to the raw client
-// provided by the operator-sdk, instead of the raw client itself.
+// NewConfiguringReconcilerWithInterfaceClient returns a configuring
+// Reconciler that reconciles an OperatorSource object in "Configuring"
+// phase. It uses the Client interface which is a wrapper to the raw
+// client provided by the operator-sdk, instead of the raw client itself.
 // Using this interface facilitates mocking of kube client interaction
 // with the cluster, while using fakeclient during unit testing.
-func NewReconcilerWithInterfaceClient(logger *log.Entry, datastore datastore.Writer, client interface_client.Client) Reconciler {
+func NewConfiguringReconcilerWithInterfaceClient(logger *log.Entry, factory appregistry.ClientFactory, datastore datastore.Writer, client interfaceClient.Client, refresher PackageRefreshNotificationSender) Reconciler {
```

Comment:
Why is it necessary to rename this function and add the word `Configuring` in it? Isn't that making the function name bigger than needed? Besides, it's an internal function, `NewConfiguringReconciler` is the function that is exposed to the factory.

Reply:
It was necessary as part of the refactor in the previous commit to distinguish between the NewReconcilerWithInterfaceClient of the Downloading phase.